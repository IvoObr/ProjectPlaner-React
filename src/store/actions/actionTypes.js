export const actionTypes = {
    START_FETCH_PROJECTS: 'START_FETCH_PROJECTS',
    START_FETCH_PROJECTS_SUCCESS: 'START_FETCH_PROJECTS_SUCCESS',
    START_FETCH_PROJECTS_FAILED: 'START_FETCH_PROJECTS_FAILED',
    CREATE_PROJECT: 'CREATE_PROJECT',
    CREATE_PROJECT_SUCCESS: 'CREATE_PROJECTS_SUCCESS',
    CREATE_PROJECT_FAILED: 'CREATE_PROJECT_FAILED',

}