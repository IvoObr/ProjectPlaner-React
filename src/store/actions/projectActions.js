import {url, requestObject, responseHandle} from '../fetchConfig'

export const createProject = (project) => {
    return(dispatch, getState) => {
        let bodyObj = {
            ...project,
        };

        requestObject.method =  'POST';
        requestObject.body = JSON.stringify(bodyObj)
        fetch(`${url}/projects`, requestObject)
            .then(response => {
                responseHandle(response, dispatch, 'CREATE_PROJECT');
            }).catch(error => {
            console.log(new Error(`Request rejected: ${error}`));
        });
    }
};

export const getProjects = () => {
    return(dispatch, getState) => {
        requestObject.method =  'GET';
        fetch(`${url}/projects`, requestObject)
            .then(response => {
                responseHandle(response, dispatch, 'GET_ALL_PROJECTS');
            }).catch(error => {
            console.log(new Error(`Request rejected: ${error}`));
        });
    }
};