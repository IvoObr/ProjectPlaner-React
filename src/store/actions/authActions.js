import {actionTypes} from "./actionTypes";

export const actions = {
  login: (user) => ({
    type: actionTypes.LOGIN_USER,
    payload: {user}
  }),
  loginSuccess: (user) => ({
    type: actionTypes.LOGIN_USER_SUCCESS,
    payload: {user}
  }),
  loginFailed: (user) => ({
    type: actionTypes.LOGIN_USER_FAILED,
    payload: {user}
  }),
  logout: (user) => ({
    type: actionTypes.LOGOUT_USER,
    payload: {user}
  }),
  logoutSuccess: (user) => ({
    type: actionTypes.LOGOUT_USER_SUCCESS,
    payload: {user}
  }),
  logoutFailed: (user) => ({
    type: actionTypes.LOGOUT_USER_FAILED,
    payload: {user}
  }),
  signup: (user) => ({
    type: actionTypes.SIGNUP_USER,
    payload: {user}
  }),
  signupSuccess: (user) => ({
    type: actionTypes.SIGNUP_USER_SUCCESS,
    payload: {user}
  }),
  signupFailed: (user) => ({
    type: actionTypes.SIGNUP_USER_FAILED,
    payload: {user}
  }),
};