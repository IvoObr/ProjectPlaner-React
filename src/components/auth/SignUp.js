import React, {Component} from 'react';
import {actions} from "../../store/actions/authActions";
import {connect} from "react-redux";
import {Redirect} from "react-router-dom";

class SignUp extends Component {
    state = {
        email: '',
        password: '',
        firstName: '',
        lastName: ''
    };

    handleChange = (event) => {
        this.setState({
            [event.target.id]: event.target.value
        })
    };

    handleSubmit = (event) => {
        event.preventDefault();
        this.props.signup(this.state);

    };

    render() {
        if (this.props.isLoggedIn) {
            return <Redirect to='/'/>
        }

        return (
            <div className='container'>
                <form onSubmit={this.handleSubmit} className='white'>
                    <h5 className="grey-text text-darken-3">Sign Up</h5>

                    <div className="input-field">
                        <label htmlFor="email">Email</label>
                        <input type="email" id='email' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <label htmlFor="password">Password</label>
                        <input type="password" id='password' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <label htmlFor="firstName">First Name</label>
                        <input type="text" id='firstName' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <label htmlFor="lastName">Last Name</label>
                        <input type="text" id='lastName' onChange={this.handleChange}/>
                    </div>


                    <div className="input-field">
                        <button className="btn green lighten-1 z-depth-0">Sign Up</button>
                    </div>
                    {this.props.loggingIn ? <div className="progress"> <div className="indeterminate"></div> </div>
                        : <div className="height-18"></div>}
                    {this.props.signUpError ?  <div className='hideElement absolute center-align'>
                        <i className="material-icons center-align large">error_outline</i>
                        <p >{this.props.signUpError}</p>
                    </div>
                        : null}
                </form>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        isLoggedIn: state.auth.loggedIn,
        loggingIn: state.auth.loggingIn,
        signUpError: state.auth.signUpError
    }
};

const mapDispatchToProps = {
    signup: actions.signup
};

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);