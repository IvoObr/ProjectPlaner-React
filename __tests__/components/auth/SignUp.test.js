import React from 'react';
import ReactDOM from 'react-dom';
import SignUp from '../../../src/components/auth/SignUp';
import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import {actions} from '../../../src/store/actions/authActions';

configure({adapter: new Adapter()});

describe('<SignUp/>', () => {
    let wrapper = null;
    let props = null;
    let store = {};
    const mockStore = configureStore();

    const initialState = {
        auth: {
            loggedIn: true,
                user: {},
            loginError: ''
        },
        project: {}
    };

        beforeEach(() => {
            store = mockStore(initialState);
            props = {
                signup: actions.signup,
                isLoggedIn: false
            };

            wrapper = shallow(<SignUp.WrappedComponent {...props} store={store}/>);
    });

    it('should render SignUp', () => {
        expect(wrapper.find('div').length).toBe(6);
    });

    it('matches snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('handleChange should work', () => {
        let event = {
            target: {
                id: 'email',
                value: 'test2'
            }
        };

        wrapper.instance().handleChange(event);
        let stateEmail = wrapper.state().email;
        expect(stateEmail).toBe('test2');
    });

    it('handleSubmit should work', () => {
        let event = {
            preventDefault() {
                return true;
            }
        };

        const throwError = wrapper.instance().handleSubmit(event);
        expect(throwError).toBeUndefined();
    });

    it('should redirect if logged in redirect to Home', () => {
        props.isLoggedIn = true;
        wrapper = shallow(<SignUp.WrappedComponent {...props}/>);
        expect(wrapper.find('Redirect').length).toBe(1);
    });
});
