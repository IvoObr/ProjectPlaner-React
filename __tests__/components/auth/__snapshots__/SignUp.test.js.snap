// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp/> matches snapshot 1`] = `
<div
  className="container"
>
  <form
    className="white"
    onSubmit={[Function]}
  >
    <h5
      className="grey-text text-darken-3"
    >
      Sign Up
    </h5>
    <div
      className="input-field"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        onChange={[Function]}
        type="email"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        minLength="8"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        htmlFor="firstName"
      >
        First Name
      </label>
      <input
        id="firstName"
        minLength="1"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        htmlFor="lastName"
      >
        Last Name
      </label>
      <input
        id="lastName"
        minLength="1"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </div>
    <div
      className="input-field"
    >
      <button
        className="btn green lighten-1 z-depth-0"
      >
        Sign Up
      </button>
    </div>
    <Loading />
    <Notifications />
  </form>
</div>
`;
